### PR TITLE
[HZ-489] Removing IMDG engine sources : Part 3

### PR DESCRIPTION
**Pull request by Fly-Style**
- Moved `schema` and `parse` packages to Jet engine.
- Moved `LogicalTableInsert` & `LogicalTableSink` to `opt/logical` package.

Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [ ] Request reviewers if possible

### CODE REVIEW

**Review by ivanthescientist — APPROVED** [review]

File: hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/schema/HazelcastTable.java [hunk 1]
```diff
@@ -50,7 +50,7 @@
 import static java.util.stream.Collectors.joining;
 
 /**
- * Base class for all tables in the Calcite integration:
+ * Base class for all IMap tables in the Calcite integration:
```
Comment (gierlachg) [inline hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/schema/HazelcastTable.java:53]: Why this change? It was correct and it is not anymore.

**Review by gierlachg — CHANGES_REQUESTED** [review]

File: hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/schema/HazelcastTable.java (hunk 1 above)
Comment (Fly-Style) [reply hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/schema/HazelcastTable.java:53]: I had wrong feeling about differences between `HazelcastTable` and `JetTable`, and tried to unify them. Then I realized  that they are different, but haven't change the comment. Thanks for catching.

**Review by gierlachg — APPROVED** [review]

**Comment by Fly-Style** [comment]
run-lab-run
